pkg/logging/rate_limited_logger.go

```diff
@@ -1,128 +1,271 @@
-	if rl.allowLog(limiter) {
-		rl.Infof(template, args...)
-	}
+	rl.logf(zapcore.InfoLevel, limiter, template, args...)
 }
 
 // Info logs the consecutive interfaces
 func (rl *RateLimitedLogger) Info(args ...interface{}) {
-	rl.Logger.Info(args...)
+	rl.logger.Info(args...)
 }
 
 // Infof logs the input template filled with argument data
 func (rl *RateLimitedLogger) Infof(template string, args ...interface{}) {
-	rl.Logger.Infof(template, args...)
+	rl.logger.Infof(template, args...)
 }
 
 // Infow logs the input message and keyValues
 func (rl *RateLimitedLogger) Infow(msg string, keysAndValues ...interface{}) {
-	rl.Logger.Infow(msg, keysAndValues...)
+	rl.logger.Infow(msg, keysAndValues...)
 }
 
 // DebugL logs a templated debug message if allowed by the rate limiter corresponding to the identifier
 func (rl *RateLimitedLogger) DebugL(limiter string, template string, args ...interface{}) {
-	if rl.allowLog(limiter) {
-		rl.Debugf(template, args...)
-	}
+	rl.logf(zapcore.DebugLevel, limiter, template, args...)
 }
 
 // Debug logs the consecutive interfaces
 func (rl *RateLimitedLogger) Debug(args ...interface{}) {
-	rl.Logger.Debug(args...)
+	rl.logger.Debug(args...)
 }
 
 // Debugf logs the input template filled with argument data
 func (rl *RateLimitedLogger) Debugf(template string, args ...interface{}) {
-	rl.Logger.Debugf(template, args...)
+	rl.logger.Debugf(template, args...)
 }
 
 // Debugw logs the input message and keyValues
 func (rl *RateLimitedLogger) Debugw(msg string, keysAndValues ...interface{}) {
-	rl.Logger.Debugw(msg, keysAndValues...)
+	rl.logger.Debugw(msg, keysAndValues...)
+}
+
+func getLogKey(limiter string, level zapcore.Level, file string, line int, payload string) string {
+	var keyWriter strings.Builder
+	keyWriter.WriteString(limiter)
+	keyWriter.WriteString("-")
+	keyWriter.WriteString(level.CapitalString())
+	keyWriter.WriteString("-")
+	keyWriter.WriteString(file)
+	keyWriter.WriteString(":")
+	keyWriter.WriteString(fmt.Sprintf("%d", line))
+	keyWriter.WriteString("-")
+	keyWriter.WriteString(payload)
+	return keyWriter.String()
+}
+
+const (
+	localFilePathPrefix = "github.com/stackrox/stackrox/"
+	filePathPrefix      = "github.com/stackrox/rox/"
+	githubPathPrefix    = "/__w/stackrox/stackrox/"
+)
+
+func getTrimmedFilePath(path string) string {
+	prefixes := []string{filePathPrefix, localFilePathPrefix, githubPathPrefix}
+	cutPath := path
+	for _, prefix := range prefixes {
+		prefixToCut := strings.Index(cutPath, prefix)
+		if prefixToCut >= 0 {
+			cutPath = cutPath[prefixToCut:]
+		}
+	}
+	trimmedPath := cutPath
+	for _, prefix := range prefixes {
+		trimmedPath = strings.TrimPrefix(trimmedPath, prefix)
+	}
+	return trimmedPath
+}
+
+func (rl *RateLimitedLogger) logf(level zapcore.Level, limiter string, template string, args ...interface{}) {
+	payload := fmt.Sprintf(template, args...)
+	_, file, line, ok := runtime.Caller(2)
```

Comment:
The file+line extraction here is to ensure we have the source location of the rate-limited log in the trace itself.
If this is not done, the log data will point at the location where the rate-limiter is sending the log, which has IMHO little added value.
```
root logger: 2023/06/30 16:35:32.319106 logger.go:78: Info: central/main.go:571 - This is a log that is issued every second to test the rate limited logger - 9 log suppressed for limiter "testlimiter"
root logger: 2023/06/30 16:35:32.320221 logger.go:78: Info: central/main.go:571 - This is a log that is issued every second to test the rate limited logger
```
Extracting and passing file+line allows to see here that the log was issued by central/main.go at line 571
Otherwise, the only available information (when using a common logger for all rate-limited logs) is logger.go:78, which I don't find very helpful.

I can check if there is available build-time information that would make this call more efficient (similar to the __FILE__ and __LINE__ macros in C). The first results I got point at `runtime.Caller`, though.